ddtrace/ddtrace.go

```diff
@@ -94,6 +98,17 @@ type SpanContext interface {
 	ForeachBaggageItem(handler func(k, v string) bool)
 }
 
+// https://github.com/open-telemetry/opentelemetry-go/blob/v1.21.0/trace/trace.go#L412
+//go:generate msgp
```

Comment:
```suggestion
//go:generate msgp -unexported -marshal=false -o=span_link_msgp.go -tests=false
```
I'd keep this the same as all of the other `//go:generate` calls in this package, unless there is a reason for it to be different?

I think this should go in its own msgp file as well, i.e. `span_link_msgp.go`

Reply:
Ahh good point. I had to modify the file remove structs we do not need to marshal. This is a lot cleaner. Is there a reason why we don't generate the tests?